src/app/src/api/boundaries.js

```diff
@@ -75,10 +81,55 @@ const boundaryApi = api.injectEndpoints({
             query: ({ id, shape }) => ({
                 url: `/boundaries/${id}/shape/`,
                 method: 'PUT',
-                data: shape,
+                data: { shape },
             }),
         }),
 
+        replaceBoundaryShape: build.mutation({
+            query: ({ id, file }) => {
+                const data = new FormData();
+                data.append('file', file, file.name);
+
+                return {
+                    url: `/boundaries/${id}/shape/`,
+                    method: 'PUT',
+                    headers: {
+                        'Content-Type': 'multipart/form-data',
+                    },
+                    data,
+                };
+            },
+            onQueryStarted: async ({ id }, { dispatch, queryFulfilled }) => {
+                const patchResult = dispatch(
+                    api.util.updateQueryData(
+                        'getBoundaryDetails',
+                        id,
+                        draftDetails => {
+                            draftDetails.submission.shape = null;
```

Comment:
Why delete the shape before we know the patch succeeds?

Reply:
This is done mainly for this:

https://github.com/azavea/iow-boundary-tool/blob/5e23e9713eaa18953b5f99a318b832b2d5684e1b/src/app/src/components/DrawTools/useEditingPolygon.js#L131-L137

We want the map to zoom to the new shape once it is loaded. If we set `hasZoomedToShape` to `false`, that `useEffect` gets called immediately and zooms to the old shape. In order for it to not fire until the new shape is loaded, we first remove the old shape.

In case the patch does not succeed, we restore the old shape in the `catch` block:

https://github.com/azavea/iow-boundary-tool/blob/b2bc04431e7916c7aec85b852c73150a169fbfe8/src/app/src/api/boundaries.js#L126-L129

Reply:
What if we moved the `dispatch(setHasZoomeToShape(false))` call to after the `queryFulfilled` promise has been resolved and the shape has been updated?

Reply:
That is what I had done originally, but it fires too quick, and still zooms to the old shape rather than the new one. Even when it is after the `dispatch(api.util.updateQueryData(...))` call.

Reply:
That makes sense. There might be a way to figure this out with async actions/extra reducers, but probably not worth getting into considering it works as is.